src/hpc/guards.h

```diff
@@ -11,109 +11,73 @@
 #ifndef GAP_HPC_GUARD_H
 #define GAP_HPC_GUARD_H
 
+#include "gasman.h"
 #include "hpc/region.h"
 #include "hpc/tls.h"
 
 #ifndef HPCGAP
 #error This header is only meant to be used with HPC-GAP
 #endif
 
-#ifdef VERBOSE_GUARDS
-void WriteGuardError(Bag bag,
-    const char *file, unsigned line, const char *func, const char *expr);
-void ReadGuardError(Bag bag,
-    const char *file, unsigned line, const char *func, const char *expr);
-#else
-void WriteGuardError(Bag bag);
-void ReadGuardError(Bag bag);
-#endif
-
-#ifdef VERBOSE_GUARDS
-#define WriteGuard(bag) WriteGuardFull(bag, __FILE__, __LINE__, __FUNCTION__, #bag)
-static ALWAYS_INLINE Bag WriteGuardFull(Bag bag,
-  const char *file, unsigned line, const char *func, const char *expr)
-#else
 static ALWAYS_INLINE Bag WriteGuard(Bag bag)
-#endif
 {
-  Region *region;
-  if (!IS_BAG_REF(bag))
+    if (!WriteCheck(bag))
+        HandleWriteGuardError(bag);
     return bag;
-  region = REGION(bag);
-  if (region && region->owner != GetTLS() && region->alt_owner != GetTLS())
-    WriteGuardError(bag
-#ifdef VERBOSE_GUARDS
-    , file, line, func, expr
-#endif
-    );
-  return bag;
 }
 
+
 EXPORT_INLINE Bag ImpliedWriteGuard(Bag bag)
 {
-  return bag;
+    return bag;
 }
 
 EXPORT_INLINE int CheckWriteAccess(Bag bag)
 {
-  Region *region;
-  if (!IS_BAG_REF(bag))
-    return 1;
-  region = REGION(bag);
-  return !(region && region->owner != GetTLS() && region->alt_owner != GetTLS())
-    || TLS(DisableGuards) >= 2;
+    Region * region;
+    if (!IS_BAG_REF(bag))
+        return 1;
+    region = REGION(bag);
+    return !(region && region->owner != GetTLS() &&
+             region->alt_owner != GetTLS()) ||
+           TLS(DisableGuards) >= 2;
 }
 
 EXPORT_INLINE int CheckExclusiveWriteAccess(Bag bag)
 {
-  Region *region;
-  if (!IS_BAG_REF(bag))
-    return 1;
-  region = REGION(bag);
-  if (!region)
-    return 0;
-  return region->owner == GetTLS() || region->alt_owner == GetTLS()
-    || TLS(DisableGuards) >= 2;
+    Region * region;
+    if (!IS_BAG_REF(bag))
+        return 1;
+    region = REGION(bag);
+    if (!region)
+        return 0;
+    return region->owner == GetTLS() || region->alt_owner == GetTLS() ||
+           TLS(DisableGuards) >= 2;
 }
 
-#ifdef VERBOSE_GUARDS
-#define ReadGuard(bag) ReadGuardFull(bag, __FILE__, __LINE__, __FUNCTION__, #bag)
-static ALWAYS_INLINE Bag ReadGuardFull(Bag bag,
-  const char *file, unsigned line, const char *func, const char *expr)
-#else
 static ALWAYS_INLINE Bag ReadGuard(Bag bag)
-#endif
 {
-  Region *region;
-  if (!IS_BAG_REF(bag))
+    if (!ReadCheck(bag))
+        HandleReadGuardError(bag);
     return bag;
-  region = REGION(bag);
-  if (region && region->owner != GetTLS() &&
-      !region->readers[TLS(threadID)] && region->alt_owner != GetTLS())
-    ReadGuardError(bag
-#ifdef VERBOSE_GUARDS
-    , file, line, func, expr
-#endif
-    );
-  return bag;
 }
 
-
```

Comment:
Why are you inserting an empty new line before `ImpliedWriteGuard`, but removing one before `ImpliedReadGuard`? I mean, I could understand if one wanted to tweak things so that always the same number of empty lines is between functions (i.e., always 1 or always 2); but instead, you seem to just switching the inconsisten spacing around?

Reply:
This was by accident, I think. Before squashing the commits, there was code that got deleted, apparently in one case without combining the lines. Fixed.